### PR TITLE
`info`: don't use a fallback for ns-qualified symbols

### PR DESCRIPTION
**Pull request by vemv**
* `info`: don't use a fallback for ns-qualified symbols
  * Closes https://github.com/clojure-emacs/orchard/issues/133
  * ℹ️ cider-nrepl will still need to get a couple tests adapted (and probably a couple more _added_)
* Add `make install` command
  * Also includes our usual refinements (`-user` etc).
* Unflake a test

### CODE REVIEW

File: src/orchard/info.clj [hunk 1]
```diff
@@ -65,32 +65,29 @@
...
-      (recur (assoc opts :ns 'clojure.core))
-      (or
-       ;; it's a special (special-symbol?)
-       (m/special-sym-meta sym)
-       ;; it's a var
-       (some-> ns (m/resolve-var sym) (m/var-meta))
-       ;; it's a Java constructor/static member symbol
-       (some-> ns (java/resolve-symbol sym))
-       ;; it's an unqualified sym maybe referred
-       (some-> ns (m/resolve-var unqualified-sym) (m/var-meta))
```
Comment (vemv) [inline src/orchard/info.clj:81]: Removed these two lines, I don't think they make sense. If one is explicitly querying `:sym foo/+`, then trying to resolve `+` without foo is not accurate.

The comment says `maybe referred`.  But `(some-> ns (m/resolve-var sym) (m/var-meta))` already handles refers, that's part of what `resolve/var`/`ns-resolve` does

File: test/orchard/info_test.clj [hunk 2]
```diff
@@ -368,7 +369,7 @@
                     (map #(select-keys % [:ns :name :arglists :macro :file]))))))
 
       (testing "- :clj"
-        (is (= (take 2 (repeat expected))
+        (is (= [{}, expected]
```
Comment (vemv) [inline test/orchard/info_test.clj:372]: Changed this expectation, `orchard.test-ns/my-add` does not exist so the right result for it is `{}`

File: test/orchard/info_test.clj [hunk 3]
```diff
@@ -467,21 +471,49 @@
...
-                        [:added :ns :name :file]))))
-  (testing "clojure.core syms can still be resolved"
-    (is (= '{:added "1.0"
-             :ns clojure.core
-             :name merge
-             :file "clojure/core.clj"}
-           (select-keys (info/info* {:ns 'gibberish :sym 'merge})
-                        [:added :ns :name :file])))))
+  (let [current-ns (-> ::_ namespace symbol)]
+    (are [input expected] (= expected
```
Comment (bbatsov) [inline test/orchard/info_test.clj:482]: I don't like `are` as I think it reduces the readability of tests. For me it's better to have more `testing` blocks, as they make it easier to follow what a group of assertions tests exactly.

File: test/orchard/java/resource_test.clj [hunk 4]
```diff
@@ -8,6 +8,8 @@
 
 (deftest project-resources-test
   (testing "get the correct resources for the orchard project"
-    (let [resources (resource/project-resources)]
+    (let [resources (->> (resource/project-resources)
```
Comment (bbatsov) [inline test/orchard/java/resource_test.clj:11]: What was wrong with this test before? There were more resources returned or something?

Comment (vemv) [reply test/orchard/java/resource_test.clj:11]: >  There were more resources returned or something?

Yes especially in local envs were there can be more resources for arbitrary reasons.

File: test/orchard/info_test.clj (hunk 3 above)
Comment (vemv) [reply test/orchard/info_test.clj:482]: Perhaps we can agree to disagree? I didn't use an `are` of the crazy flavor here. It's a simple mapping of inputs->outputs.

Normally to reach this level of exhaustiveness with manually coded tests, one would have to take 1 to 2 'screenfuls' worth of code. That's generally not too readable - it's so wordy that it's hard to compare the test cases between them.

Normally people give up before even attempting this level of exhaustiveness, so `are` can invite us to do the right thing (focus on the data, compare the tests cases between them, cheaply add more cases if missing)

Comment (bbatsov) [reply test/orchard/info_test.clj:482]: I can live with this, I simply don't like it. Too many years of writing RSpecs I guess. :-) There's also the practical problem that CIDER's test runner doesn't work well with `are`, but that's not a big deal.

File: test/orchard/java/resource_test.clj (hunk 4 above)
Comment (bbatsov) [reply test/orchard/java/resource_test.clj:11]: Got it.

**Comment by vemv** [comment]
0.7.3 released with these changes and consumed in cider-nrepl https://github.com/clojure-emacs/cider-nrepl/pull/715

**Comment by plexus** [comment]
Thanks for picking this up so quickly!

**Comment by vemv** [comment]
🍻 Turned out that https://github.com/clojure-emacs/orchard/commit/7d7f752e5d1abd1cdfbbcd164a7a5ca955d7789d wasn't wrong per se but it helped surface various edge cases (covered with tests here and downstream)
